src/main/java/winstone/HostConfiguration.java

```diff
@@ -230,10 +232,12 @@ protected File getWebRoot(File requestedWebroot, File warfile) throws IOExceptio
                 File tempFile = File.createTempFile("dummy", "dummy");
                 String userName = System.getProperty("user.name");
                 unzippedDir = new File(tempFile.getParent(),
-                        (userName != null ? WinstoneResourceBundle.globalReplace(userName, 
+                        (userName != null ? WinstoneResourceBundle.globalReplace(userName,
                                 new String[][] {{"/", ""}, {"\\", ""}, {",", ""}}) + "/" : "") +
                         "winstone/" + warfile.getName());
-                tempFile.delete();
+                if (!tempFile.delete()) {
```

Comment:
Or there is a Commons IO utility, or you can use `java.nio.file.Files.delete` which throws a meaningful exception.